modules/goat/taxonsearch/main.nf

```diff
@@ -0,0 +1,36 @@
+process GOAT_TAXONSEARCH {
+    tag "$meta.id"
+    label 'process_low'
+
+    conda (params.enable_conda ? "bioconda::goat=0.2.0" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/goat:0.2.0--h92d785c_0':
+        'quay.io/biocontainers/goat:0.2.0--h92d785c_0' }"
+
+    input:
+    tuple val(meta), val(taxon), path(taxa_file)
+
+    output:
+    tuple val(meta), path("*.tsv"), emit: taxonsearch
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    input = taxa_file.name != 'NO_FILE' ? "-f ${taxa_file}" : "-t ${taxon}"
```

Comment:
If I understand correctly, you are trying to determine whether a file is entered or not? If that's the case you can add an empty channel using `[]` as input and chance your check to `input = taxa_file.name ? "-f ${taxa_file}" : "-t ${taxon}"`. This is the default way on how to specify an optional channel :)

Reply:
Same for the next two lines